insights/combiners/redhat_release.py

```diff
@@ -61,4 +66,72 @@ def redhat_release(rh_release, un):
     if rh_release:
         return Release(rh_release.major, rh_release.minor)
 
-    raise Exception("Unabled to determine release.")
+    raise ParseException("Unabled to determine release.")
+
+
+@combiner([Uname, rht_release])
+class RedHatRelease(object):
+    """
+    Combiner class to check uname and redhat-release for RHEL major/minor
+    version.
+    Prefer uname to redhat-release.
+
+    Attributes:
+        major (int): The major RHEL version.
+        minor (int): The minor RHEL version.
+        rhel (str): The RHEL version, e.g. '7.2', '7.5-0.14'
+        rhel6 (str): The RHEL version when it's RHEL6, otherwise None
+        rhel7 (str): The RHEL version when it's RHEL7, otherwise None
+        rhel8 (str): The RHEL version when it's RHEL8, otherwise None
+
+    Raises:
+        ParseException: If the version can't be determined even though a Uname
+            or RedhatRelease was provided.
+
+    Examples:
+        >>> rh_rel.rhel
+        '7.2'
+        >>> rh_rel.major
+        7
+        >>> rh_rel.minor
+        2
+        >>> rh_rel.rhel6 is None
+        True
+        >>> rh_rel.rhel7
+        '7.2'
+        >>> rh_rel.rhel8 is None
+        True
+    """
+    def __init__(self, uname, rh_rel):
+        self.major = self.minor = None
+        self.rhel = self.rhel6 = self.rhel7 = self.rhel8 = None
+        if uname and uname.redhat_release.major != -1:
+            self.major = uname.redhat_release.major
+            self.minor = uname.redhat_release.minor
+            self.rhel = '{0}.{1}'.format(self.major, self.minor)
+        elif rh_rel and rh_rel.is_rhel:
+            self.major = rh_rel.major
+            self.minor = rh_rel.minor
+            self.rhel = rh_rel.version
+
+        if self.rhel is None:
+            raise ParseException("Unable to determine release.")
+
+        self.rhel6 = self.rhel if self.major == 6 else None
+        self.rhel7 = self.rhel if self.major == 7 else None
+        self.rhel8 = self.rhel if self.major == 8 else None
+
+
+# No need to serialize/deserialize the `rhel#` attributes
+@serializer(RedHatRelease)
+def serialize_RedHatRelease(obj, root=None):
+    return {"major": obj.major, "minor": obj.minor, "rhel": obj.rhel}
```

Comment:
All attributes of the combiner instance that could be accessed by a rule need to be serialized and deserialized or there will be attribute errors since the deserializer bypasses `__init__`. Need to add rhel[6-8].

Reply:
I got it. Thanks for the explanation. @csams